substrate/frame/support/src/lib.rs

```diff
@@ -1111,23 +1111,37 @@ pub mod pallet_macros {
 	/// #
 	/// 	#[pallet::pallet]
 	/// 	pub struct Pallet<T>(_);
+	///
+	/// # 	#[pallet::config]
+	/// # 	pub trait Config: frame_system::Config {
+    ///         /// Type for specifying dispatchable weights.
+    ///         type WeightInfo: WeightInfo;
+    ///     }
 	///
 	/// 	#[pallet::call]
 	/// 	impl<T: Config> Pallet<T> {
-	/// 		#[pallet::weight({0})] // <- set actual weight here
+	/// 		#[pallet::weight(<T as Config>::WeightInfo::do_something())] 
```

Comment:
The issue was talking about the following syntax:

```
#[pallet::weight(<T as Config>::WeightInfo)]
```

But your example also makes sense! It would be nice if you could include both examples :)

Reply:
I already added examples using the `#[pallet::weight(<T as Config>::WeightInfo)]` or `#[pallet::call(weight = <T as Config>::WeightInfo)]` attribute.  which examples do you mean, please?